substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/service/HasGcNotificationSupport.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, 2024, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.genscavenge.service;
+
+import java.util.function.BooleanSupplier;
+
+import com.oracle.svm.core.VMInspectionOptions;
+
+public class HasGcNotificationSupport implements BooleanSupplier {
+    @Override
+    public boolean getAsBoolean() {
+        return VMInspectionOptions.hasGcNotificationSupport();
```

Comment:
Just a note: I was hoping that we could include this functionality unconditionally but it seems that this is not an option because it pulls in too much JDK code.